extensions/ql-vscode/src/view/variant-analysis/VariantAnalysisHeader.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import { VariantAnalysisStatus } from '../../remote-queries/shared/variant-analysis';
+import ViewTitle from '../remote-queries/ViewTitle';
+import { CodeSquareIcon, FileCodeIcon } from '@primer/octicons-react';
+import { LinkIconButton } from './LinkIconButton';
+import styled from 'styled-components';
+import { VSCodeButton } from '@vscode/webview-ui-toolkit/react';
+
+export type VariantAnalysisHeaderProps = {
+  queryName: string;
+  queryFileName: string;
+  status: VariantAnalysisStatus;
+
+  onOpenQueryClick: () => void;
+  onViewQueryClick: () => void;
+
+  onStopQueryClick: () => void;
+
+  onCopyRepositoryListClick: () => void;
+  onExportResultsClick: () => void;
+};
+
+const Container = styled.div`
+  display: flex;
+  align-items: center;
+`;
+
+const TopRow = styled.div`
+  max-width: 100%;
+`;
+
+const TopLinkButtons = styled.div`
+  display: flex;
+  gap: 1em;
+`;
+
+const BottomButtons = styled.div`
+  margin-left: auto;
+  display: flex;
+  gap: 1em;
+`;
+
+const Button = styled(VSCodeButton)`
+  white-space: nowrap;
+`;
+
+export const VariantAnalysisHeader = ({
```

Comment:
So are you imaging this component to have the whole header area (as described in Figma) or just what we have now? I think the size of what you have now is big enough and ideally shouldn't get too much bigger.

Reply:
Perhaps the fact that we can't come up with a good name is an indication that the component is doing a bit too much. I think that's also shown by the fact that we need to pass quite a few inputs.

It's currently doing two things:
1. Allows users to see query information (title and links to query file/text).
2. Allows users to execute certain actions against a variant analysis.

Perhaps we should split into two components - `QueryDetails` and `VariantAnalysisActions` or something like that?

Reply:
👍, I think that makes sense. The actions were already split into a separate component, so now I've also split out the query header into a separate component. I've kept the `VariantAnalysisHeader` component, but it now just contains a container and the two components. We can also hopefully add the upcoming stats component to this without the file becoming too large.

I haven't created stories for the two separate components since that might result in stories which are too granular. I'm also not sure whether we want to create separate unit tests for these two components or whether we want to keep the current unit tests on the parent component. I have now added some tests for the actions since those actually contain logic, but have not added any for the query details.

Reply:
If you were starting fresh, would you create unit tests and stories against the individual components or the top level one? I think we'd do the smaller components. 

We potentially don't even need the `VariantAnalysisHeader` component - at the moment it feels like it just adds to prop drilling. But I don't have a strong sense about that yet.

Reply:
Good point, thanks! I've added stories and tests for the smaller components and removed the tests for the `VariantAnalysisHeader` component. I still see value in having the `VariantAnalysisHeader` and the stories around for now while we're still working on these components to see how everything fits together.

I definitely think we should refactor the `VariantAnalysisHeader` component (probably just remove it or change its props to take a complete variant analysis interface once we have that). However, I think that will be easier once we actually have an interface from which we can get the data, so I've left it like this for now.

Reply:
Awesome thanks!